dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs

```diff
@@ -98,25 +98,25 @@ public async Task<Plan> CreatePlanAsync(string goal)
         }
 
         // Build and return plan
-        ISKFunction function;
+        var plan = new Plan(goal);
         if (planData.Plan.Function.Contains("."))
         {
             var parts = planData.Plan.Function.Split('.');
-            function = this._context.Skills!.GetFunction(parts[0], parts[1]);
+            plan.AddSteps(this._context.Skills!.GetFunction(parts[0], parts[1]));
         }
-        else
+        else if (!string.IsNullOrWhiteSpace(planData.Plan.Function))
         {
-            function = this._context.Skills!.GetFunction(planData.Plan.Function);
+            plan.AddSteps(this._context.Skills!.GetFunction(planData.Plan.Function));
```

Comment:
actually I think we should store the function in the root. There was a bug in the plan object forcing us to use AddStep. Now we can use:

```
var plan = new Plan(function)
```

(no need to store the goal)

Reply:
if you like, we need a Plan ctor that allows to pass a goal and a single function, storing the function in the root, ie something like

```csharp
public Plan(ISKFunction function, string goal, string? functionName = null, string? skillName = null)
{
    this.SetFunction(function);
    this.Description = goal;
    this.Name = functionName ?? goal;
    this.SkillName = skillName ?? this.GetType().FullName;
}
```

/cc @lemillermicrosoft

Reply:
We have `public Plan(ISKFunction function)` and `public Plan(string goal, params ISKFunction[] steps)`. Maybe we can use one of those. I'd like to think more about a single function being the root of a plan and the result of overwriting that functions description with a goal.